src/components/projekti/kansalaisnakyma/PalauteLomakeDialogi.tsx

```diff
@@ -0,0 +1,286 @@
+import { DialogActions, DialogContent } from "@mui/material";
+import React, { ReactElement, useCallback, useState } from "react";
+import Button from "@components/button/Button";
+import HassuStack from "@components/layout/HassuStack";
+import HassuDialog from "@components/HassuDialog";
+import HassuGrid from "@components/HassuGrid";
+import { FormProvider, useForm, UseFormProps, Controller, FieldError } from "react-hook-form";
+import { palauteSchema } from "src/schemas/vuorovaikutus";
+import { yupResolver } from "@hookform/resolvers/yup";
+import useTranslation from "next-translate/useTranslation";
+import { VuorovaikutusJulkinen, ProjektiJulkinen, PalauteInput, api } from "@services/api";
+import { formatDate } from "src/util/dateUtils";
+import TextInput from "@components/form/TextInput";
+import Textarea from "@components/form/Textarea";
+import CheckBox from "@components/form/CheckBox";
+import IconButton from "@components/button/IconButton";
+import FormGroup from "@components/form/FormGroup";
+import axios from "axios";
+import HassuSpinner from "@components/HassuSpinner";
+import useSnackbars from "src/hooks/useSnackbars";
+import log from "loglevel";
+
+interface Props {
+  open: boolean;
+  onClose: () => void;
+  vuorovaikutus: VuorovaikutusJulkinen;
+  projekti: ProjektiJulkinen;
+}
+
+interface PalauteFormInput {
+  etunimi: string;
+  sukunimi: string;
+  sahkoposti: string;
+  puhelinnumero: string;
+  kysymysTaiPalaute: string;
+  yhteydenottotapaEmail: boolean | null;
+  yhteydenottotapaPuhelin: boolean | null;
+  liite: string | null;
+}
+
+const defaultValues = {
+  kysymysTaiPalaute: "",
+  yhteydenottotapaEmail: false,
+  yhteydenottotapaPuhelin: false,
+  liite: null
+};
+
+export default function PalauteLomakeDialogi({ open, onClose, projekti, vuorovaikutus }: Props): ReactElement {
+  const { t } = useTranslation();
+  const [tiedosto, setTiedosto] = useState<File | undefined>(undefined);
+  const [formIsSubmitting, setFormIsSubmitting] = useState(false);
+  const [kiitosDialogiOpen, setKiitosDialogiOpen] = useState(false);
+
+  const formOptions: UseFormProps<PalauteFormInput> = {
+    resolver: yupResolver(palauteSchema, { abortEarly: false, recursive: true }),
+    mode: "onChange",
+    reValidateMode: "onChange",
+    defaultValues
+  };
+  const { showSuccessMessage, showErrorMessage } = useSnackbars();
+  const useFormReturn = useForm<PalauteFormInput>(formOptions);
+
+  const {
+    register,
+    handleSubmit,
+    control,
+    formState: { errors },
+    setValue,
+    reset
+  } = useFormReturn;
+
+  const talletaTiedosto = useCallback(async (tiedosto: File) => {
```

Comment:
Pitikö liitteen koko rajoittaa tekstin mukaisesti 4.5Mt? Nyt ainakin 10Mt liitteetkin menevät läpi.

Reply:
Tein rajoituksen, ja se näkyy myös käyttäjälle. BE-puolta en toki toteuttanut.